### ISSUE TITLE
Per instance vertex color

### ISSUE THREAD

**Issue by tomankirilov** [godotengine/godot-proposals#6486]
### Describe the project you are working on

A vertex painting addon for godot 4.x

### Describe the problem or limitation you are having in your project

I am currently editing the mesh itself and that results in mesh duplication.
That makes reimporting the mesh not reimporting the mesh with edited vertex colors.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Unity or Unreal engine save the vertex data per instance. That way the mesh remains the same and after reimport the vertex color is not lost. Sometimes it's scrambled if the mesh is too different from the original but in most cases edits are not chaning vertex count or order.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

A simple way would be for the MeshInstance3D to save the vertex data and not the mesh itself.
Certainly there are better ways to go about it ...

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It could be, but it will be ugly and not performant. The best way would be to be a part of either the MeshInstance3D or the mesh resource itself.

### Is there a reason why this should be core and not an add-on in the asset library?

In my experience as an environment artist I have worked with multiple engines. Both publicly available and in-house.I never worked on a 3d project that does not require vertex color for some aspect of the workflow.

**Comment by Calinou** [comment]
This will require something similar to UV2 import cache to be implemented. Given the editor does not alter vertex colors, I doubt this will be done anytime soon.

**Comment by solitaryurt** [comment]
@Calinou, agreed. This is definitely something the engine is missing. This would enable a whole new set of plugins for the engine that don't currently exist.

**Comment by solitaryurt** [comment]
External bin that's a resource makes sense, allows for you to have different resources for the same mesh. Omitting the resource is simple as well.

**Comment by tomankirilov** [comment]
Is there any chance that this may be implemented in the engine relatively soon?
I've been working on an addon for editing vertex colors since version 3.x.
I started rewriting it for 4.x but I realized that for most use cases per instance vertex color is required.

**Comment by Calinou** [comment]
> Is there any chance that this may be implemented in the engine relatively soon? I've been working on an addon for editing vertex colors since version 3.x. I started rewriting it for 4.x but I realized that for most use cases per instance vertex color is required.

To my knowledge, nobody is currently working on this and 4.1 is in feature freeze. As a result, we don't expect this to be implemented anytime soon.
